crates/holochain_conductor_api/src/config/conductor.rs

```diff
@@ -224,12 +234,134 @@ impl Default for NetworkConfig {
+        ) -> ConductorConfigResult<()> {
+            if let Some(module_config) = module_config.get_mut(module) {
+                if let Some(module_config) = module_config.as_object_mut() {
+                    if module_config.contains_key(key) {
+                        tracing::warn!("The {} module configuration contains a '{}' field, which is being overwritten", module, key);
+                    }
+
+                    // The config for this module exists and is an object, insert the key-value pair
+                    module_config.insert(key.into(), value);
+                } else {
+                    // The configuration for this module exists, but isn't an object
+                    return Err(ConductorConfigError::InvalidNetworkConfig(format!(
+                        "advanced.{} field must be an object",
+                        module
+                    )));
+                }
+            } else {
+                // The config for this module isn't set at all, so we need to insert it
+                module_config.insert(
+                    module.into(),
+                    serde_json::json!({
+                        key: value,
+                    }),
+                );
+            }
+
+            Ok(())
+        }
+
+        if let Some(module_config) = working.as_object_mut() {
+            insert_module_config(
+                module_config,
+                "coreBootstrap",
+                "serverUrl",
+                serde_json::Value::String(self.bootstrap_url.as_str().into()),
+            )?;
+
+            insert_module_config(
+                module_config,
+                "tx5Transport",
+                "serverUrl",
+                serde_json::Value::String(self.signal_url.as_str().into()),
+            )?;
+
+            if let Some(initiate_interval_ms) = self.gossip_initiate_interval_ms {
+                insert_module_config(
+                    module_config,
+                    "k2Gossip",
+                    "initiateIntervalMs",
+                    serde_json::Value::Number(serde_json::Number::from(initiate_interval_ms)),
+                )?;
+            }
+
+            if let Some(min_initiate_interval_ms) = self.gossip_min_initiate_interval_ms {
+                insert_module_config(
+                    module_config,
+                    "k2Gossip",
+                    "minInitiateIntervalMs",
+                    serde_json::Value::Number(serde_json::Number::from(min_initiate_interval_ms)),
+                )?;
+            }
+
+            if let Some(round_timeout_ms) = self.gossip_round_timeout_ms {
+                insert_module_config(
+                    module_config,
+                    "k2Gossip",
+                    "roundTimeoutMs",
+                    serde_json::Value::Number(serde_json::Number::from(round_timeout_ms)),
+                )?;
+            }
+
+            // TODO nowhere to put the webrtc config in K2 yet!
```

Comment:
huh. woops!

Reply:
Haha, all part of the fun of integrating all of this. We've mostly done a good job of preparing things to line up, if this is the kind of thing we're dealing with them I'm very happy :D